crates/ruff/src/rules/flake8_simplify/rules/ast_with.rs

```diff
@@ -60,9 +60,15 @@ impl Violation for MultipleWithStatements {
     }
 }
 
-fn find_last_with(body: &[Stmt]) -> Option<(&[Withitem], &[Stmt])> {
-    let [Stmt::With(ast::StmtWith { items, body, .. })] = body else { return None };
-    find_last_with(body).or(Some((items, body)))
+/// Returns a boolean indicating its an async with statement, the items and
+/// body.
+fn find_last_with(body: &[Stmt]) -> Option<(bool, &[Withitem], &[Stmt])> {
```

Comment:
sorry, bad example. If i use
```python
async with b as b2:
    with c as c2:
        async with d as d2:
            f(b2, c2, d2)
```
i get 
```
SIM117.py:112:1: SIM117 [*] Use a single `with` statement with multiple contexts instead of nested `with` statements
    |
110 |       return 0
111 |   
112 | / async with b as b2:
113 | |     with c as c2:
114 | |         async with d as d2:
    | |___________________________^ SIM117
115 |               f(b2, c2, d2)
    |
    = help: Combine `with` statements

ℹ Suggested fix
109 109 | 
110 110 |     return 0
111 111 | 
112     |-async with b as b2:
113     |-    with c as c2:
114     |-        async with d as d2:
115     |-            f(b2, c2, d2)
    112 |+async with b as b2, c as c2:
    113 |+    async with d as d2:
    114 |+        f(b2, c2, d2)

```
which is incorrect.

Could you add a test case for both the already working example and this example?

Reply:
Yeah if they're out-of-order (with, then async, then with, etc.), we shouldn't re-order the context managers IMO -- only consecutive sync or async managers.